### ISSUE TITLE
Does pangenome pipeline rely on softmask assemblies?

### ISSUE THREAD

**Issue by baozg** [ComparativeGenomicsToolkit/cactus#733]
Hi, 

We have dozens of plant HiFi assemblies of inbreeding line, my previous test work on pggb for parameters tuning. I noticed the newest version of cactus including updates of pangenome pipeline. It no longer need dna-brnn for masking centromere first, which cannot running in the non-human assemblies.  So for running pipeline of same specices (low-diversity), do I need softmask the assemblies first? (using annotation from RepeatMasker)

**Comment by glennhickey** [comment]
You do not need to do any masking for the pangenome pipeline (as of version 2.1.0).  Any softmasking in your inputs will be ignored.  I haven't tried any plants yet, but if the divergence is low enough for minigraph to run, you should get good results.  Please post here if you run into any issues.

**Comment by baozg** [comment]
I run `Minigraph-Cactus` succeffully. I wonder how do you define the order of `Minigraph` in the HPRC? Minigraph shoulde be order-bias. And are there any parameters I can set to get higher sensitivity?

All run use the default parameters

```shell

cactus-minigraph ./jobstore_${chr} Pan.${chr}.pg.txt Pan23.${chr}.gfa.gz --realTimeLogging --reference S --mapCores 64 --maxCores 64 --workDir minigraph-cactucs
cactus-graphmap ./jobstore_${chr} Pan.${chr}.pg.txt Pan23.${chr}.gfa.gz Pan23.${chr}.paf --reference S --maxCores 64 --mapCores 8 --defaultCores 64 --workDir minigraph-cactucs --outputFasta Pan23.${chr}.gfa.fa.gz
cactus-align ./jobstore_${chr} Pan.${chr}.pg.txt Pan23.${chr}.paf Pan23.${chr}.hal --pangenome --pafInput --outVG --reference S --maxCores 64 --consCores 8 --defaultCores 64 --workDir minigraph-cactucs
cactus-graphmap-join ./jobstore_${chr} --vg Pan23.${chr}.vg --outDir ./ --outName Pan.${chr} --reference S --vcf --gfaffix --wlineSep "." --realTimeLogging
```

**Comment by glennhickey** [comment]
Minigraph is definitely order-dependent.  The most important choice is the first genome to add, which is specified by `--reference` in the minigraph-cactus pipeline.  For HPRC, I believe the order was GRCh38, CHM13, then everything else in lexicographic order by sample name.  (for a CHM13-referenced graph, it would be CHM13, GRCh38, then everything else). 

I could imagine heuristics for trying to optimize the order, but it's a large, expensive search space so I don't think much work has been done on it -- and I don't think it would have too much of a practical effect.  

For making minigraph more sensitive, you can tune the parameters here:
https://github.com/ComparativeGenomicsToolkit/cactus/blob/v2.1.0/src/cactus/cactus_progressive_config.xml#L319-L320
(if all contigs have unique names, you can also run minigraph directly to make your graph).  For suggestions on particular values to try in order to increase sensitivity, you might try asking [here](https://github.com/lh3/minigraph/issues)

**Comment by baozg** [comment]
Thank you for promptly response. I remember a cattle pangenome paper using `mash distance` based tree for estimation order. Using a tree order for minigraph maybe make sense. 

But for now I more want to know how to run cactus with more sensetivity?

**Comment by glennhickey** [comment]
I agree that something like mash distance would be interesting to explore for minigraph ordering. 

The sensitivity of the Minigraph-Cactus pangenome pipeline will be almost entirely bound by minigraph.  The cactus part is pretty much the same whether your making pangenomes or using Progressive Cactus to align human to mouse.  It uses abPOA to compute multiple alignments (and this part does use mash distance to determine ordering of the partial order alignment) .
